### PR TITLE
Add internal text document model

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
+/*-
+ * *
+ * *
+ * Copyright (C) 2020 - 2021 the original author or authors.
+ * *
```
Comment (valfirst) [inline vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java:5]: something wrong here

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            }
+        });
+    }
+
+    public List<String> getTextDocument(String identifier)
+    {
+        return new ArrayList<>(textDocuments.get(identifier));
+    }
+
+    private boolean apply(TextDocumentContentChangeEvent event, List<String> container)
```
Comment (valfirst) [inline vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java:80]: `apply` -> `handleEvent`

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+        if (startPos.getLine() < endPos.getLine())
+        {
+            String endLine = container.get(endPos.getLine());
+
+            insertLines(List.of(positionalConcat(startLine, startPos, endLine, endPos, eventText)), container, startPos,
+                    endPos);
+            return true;
+        }
+        else if (startPos.getLine() == endPos.getLine())
```
Comment (valfirst) [inline vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java:128]: `else` is redundant

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+import org.eclipse.lsp4j.TextDocumentContentChangeEvent;
+import org.eclipse.lsp4j.TextDocumentItem;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Singleton
+public class TextDocumentEditor implements TextDocumentEventListener
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(TextDocumentEditor.class);
+    private final Map<String, List<String>> textDocuments = new HashMap<>();
```
Comment (ikalinin1) [inline vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java:47]: I propose to clean-up during onClose event

**Review by ikalinin1 — APPROVED** [review]

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java (hunk 4 above)
Comment (uarlouski) [reply vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java:48]: fixed

**Pull request merged**